Add unit test for SequentialReader when metadata file does not exist

zmichaels11 opened the pull request:
### Changes
* ~~Pass `storage_options.storage_id` to `open_storage` when metadata file doesn't exist~~ Resolved in https://github.com/ros2/rosbag2/pull/257
* ~~Add line break in `compression_options` since linters were complaining about line length.~~ Resolved in https://github.com/ros2/rosbag2/pull/256
* Add unit test to `rosbag_cpp` checking if `storage_id` is resolved when `metadata_file` does not exist.

### Issues
* #253 

Signed-off-by: Zachary Michaels <zmichaels11@gmail.com>

piraka9011 commented on rosbag2_cpp/test/rosbag2_cpp/test_storage_without_metadata_file.cpp:
```diff
@@ -0,0 +1,88 @@
+#include "mock_metadata_io.hpp"
+#include "mock_storage.hpp"
+#include "mock_storage_factory.hpp"
+using namespace testing;  // NOLINT
```
Nit: Spacing

zmichaels11 commented:
@ros2/aws-oncall - please run this CI job
Gist: https://gist.githubusercontent.com/zmichaels11/543a323a0b3788054a44899fc4eecfd5/raw/48a24b6fa82faaca8463a92fbffc594f0bfc7d19/ros2.repos
BUILD args: --packages-up-to rosbag2_cpp
TEST args: --packages-select rosbag2_cpp
Job: ci_launcher

prajakta-gokhale commented:
CI with `rosbag2_cpp` and `rosbag2_bag_v2`:

* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=9027)](http://ci.ros2.org/job/ci_linux/9027/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=4786)](http://ci.ros2.org/job/ci_linux-aarch64/4786/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=7387)](http://ci.ros2.org/job/ci_osx/7387/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=8961)](http://ci.ros2.org/job/ci_windows/8961/)

zmichaels11 commented:
`rosbag2_bag_v2` isn't on the repos file. I'll add it manually

zmichaels11 commented:
I updated the gist (see above) to include the `rosbag2_bag_v2` fix
I also cancelled the in-flight tests because macOS was failing due to unused variable. (now removed)

@ros2/aws-oncall - please run this CI job
Gist: https://gist.githubusercontent.com/zmichaels11/543a323a0b3788054a44899fc4eecfd5/raw/b983d85e735c80a1cca3ab36e3a3ae2839c07224/ros2.repos
BUILD args: --packages-up-to rosbag2_cpp rosbag2_bag_v2
TEST args: --packages-select rosbag2_cpp
Job: ci_launcher

prajakta-gokhale commented:
CI again for `rosbag2_cpp` and `rosbag2_bag_v2`:

* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=9029)](http://ci.ros2.org/job/ci_linux/9029/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=4788)](http://ci.ros2.org/job/ci_linux-aarch64/4788/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=7389)](http://ci.ros2.org/job/ci_osx/7389/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=8963)](http://ci.ros2.org/job/ci_windows/8963/)

zmichaels11 commented:
I made a mistake, CI needs to run with packages-up-to `rosbag2_bag_v2_plugins`

Karsten1987 commented:
You'd need to start a packaging job to test the `rosbag2_bag_v2_plugin` as it requires the ros1_bridge and therefore melodic to be sourced beforehand.

zmichaels11 commented:
@thomas-moulard do we have the ability to do packaging job launches?

zmichaels11 commented:
Packaging CI:
[![Build Status](https://ci.ros2.org/job/ci_packaging_linux/283/badge/icon)](https://ci.ros2.org/job/ci_packaging_linux/283/)

zmichaels11 commented:
The changes to `rosbag2_cpp` introduced to this PR were merged in with PR #257.
This PR is now just a new e2e test so I'll run packaging on the issue over in `rosbag2_bag_v2`.

@ros2/aws-oncall  - please run this CI job
Gist: https://gist.githubusercontent.com/zmichaels11/13732b60becb45252e8c8f5796267158/raw/48a24b6fa82faaca8463a92fbffc594f0bfc7d19/ros2.repos
BUILD args: --packages-up-to rosbag2_cpp
TEST args: --packages-select rosbag2_cpp
Job: ci_launcher

piraka9011 commented:
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=9062)](http://ci.ros2.org/job/ci_linux/9062/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=4811)](http://ci.ros2.org/job/ci_linux-aarch64/4811/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=7415)](http://ci.ros2.org/job/ci_osx/7415/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=8993)](http://ci.ros2.org/job/ci_windows/8993/)
